orttraining/orttraining/eager/opgen/opgen/generator.py

```diff
@@ -388,11 +377,36 @@ def _write_function_body(
       writer.write(f'std::vector<OrtValue> {onnx_op.outputs}')
       writer.writeline(f'({onnx_op.outputs.count});')
 
-      if in_place_param:
-        assert(onnx_op.outputs.count == 1)
-        # TODO: This assumes that the first output corresponds to the first input.
-        # This may not work for more complicated ops.
-        writer.writeline(f'{onnx_op.outputs}[0] = ort_input_{onnx_op.inputs[0]};')
+      return_info = cpp_func.torch_func.return_type if cpp_func.torch_func else None
+      in_place_params = {}
+
+      if return_info:
+        for input_index, op_input in enumerate(onnx_op.inputs):
+          if isinstance(op_input, Outputs):
+            continue
+        
+          # See if this input is aliased as an in-place tensor
+          cpp_param = cpp_func.get_parameter(op_input)
+          if cpp_param:
+            for torch_p in cpp_param.torch_param:
+              if isinstance(return_info, ast.TupleType):
```

Comment:
I am not sure whether it is good idea to use tuple in the return type. there is some type like tensor list or array used in Aten native function. could we use those types?

Reply:
All the native ATen functions which return multiple in-place ops use std::tuple. e.g.
::std::tuple<Tensor &,Tensor &> cummax_out(const Tensor & self, int64_t dim, Tensor & values, Tensor & indices); // {"schema": "aten::cummax.out(Tensor self, int dim, *, Tensor(a!) values, Tensor(b!) indices) -> (Tensor(a!) values, Tensor(b!) indices)", "dispatch": "True", "default": "True"}